controllers/azuremachine_reconciler.go

```diff
@@ -94,6 +96,24 @@ func (s *azureMachineService) reconcile(ctx context.Context) error {
 	return nil
 }
 
+// pause pauses all components making up the machine.
+func (s *azureMachineService) pause(ctx context.Context) error {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureMachineService.Pause")
```

Comment:
```suggestion
	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureMachineService.pause")
```

Reply:
I noticed this was capitalized for both `reconcile` and `delete` as well. Should I update those to be lowercase too?

Reply:
I think so, because aren't we trying to identify the literal name of the function so we can match it up when looking at telemetry?

It's a bit confusing here because you're actually assigning `pause` to `ams.Pause`, and the exported name feels like it should be more visible...or something. But I think the literal name is correct here.